bin/run.py

```diff
@@ -40,35 +47,54 @@ def run(self, bar, *, interaction=None, submission_args=None):
         else:
-            result = self.submission.run(self.testcase.in_path, self.out_path)
-            if result.duration > self.problem.settings.timelimit:
-                result.verdict = 'TIME_LIMIT_EXCEEDED'
-                if result.timeout_expired:
-                    result.print_verdict_ = 'TLE (aborted)'
-            elif result.status == ExecStatus.ERROR:
-                result.verdict = 'RUN_TIME_ERROR'
-                if config.args.error:
-                    result.err = 'Exited with code ' + str(result.returncode) + ':\n' + result.err
-                else:
-                    result.err = 'Exited with code ' + str(result.returncode)
-            else:
-                # Overwrite the result with validator returncode and stdout/stderr, but keep the original duration.
-                duration = result.duration
-                result = self._validate_output(bar)
-                if result is None:
-                    error(f'No output validators found for testcase {self.testcase.name}')
-                    result = ExecResult(None, ExecStatus.REJECTED, 0, False, None, None)
-                    result.verdict = 'VALIDATOR_CRASH'
-                else:
-                    result.duration = duration
-
-                    if result.status:
-                        result.verdict = 'ACCEPTED'
-                    elif result.status == ExecStatus.REJECTED:
-                        result.verdict = 'WRONG_ANSWER'
+            nextpass = self.feedbackdir / 'nextpass.in' if self.problem.multipass else None
+            while True:
+                result = self.submission.run(self.in_path, self.out_path)
+                if result.duration > self.problem.settings.timelimit:
+                    result.verdict = 'TIME_LIMIT_EXCEEDED'
+                    if result.timeout_expired:
+                        result.print_verdict_ = 'TLE (aborted)'
+                elif result.status == ExecStatus.ERROR:
+                    result.verdict = 'RUN_TIME_ERROR'
+                    if config.args.error:
+                        result.err = (
+                            'Exited with code ' + str(result.returncode) + ':\n' + result.err
+                        )
                     else:
-                        config.n_error += 1
+                        result.err = 'Exited with code ' + str(result.returncode)
+                else:
+                    # Overwrite the result with validator returncode and stdout/stderr, but keep the original duration.
+                    duration = result.duration
+                    result = self._validate_output(bar)
+                    if result is None:
+                        error(f'No output validators found for testcase {self.testcase.name}')
+                        result = ExecResult(None, ExecStatus.REJECTED, 0, False, None, None)
                         result.verdict = 'VALIDATOR_CRASH'
+                    else:
+                        result.duration = duration
```

Comment:
Take a `max` here?

Reply:
ahm I didn't change this... But I also think this is right. Why should the validator run time be relevant here?